### PR TITLE
Feature users

### PR DESCRIPTION
**Pull request by Minuta18**
Global system which provides operations with users. There is no any auth yet.

### CODE REVIEW

**Comment by M-Stepan-373** [comment]
Its gut and schon functions

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,6 @@ nginx-service/var/
 postgres-admin-tar/
 postgres-service/data/
 postgres-admin-service/data/
-.venv
\ No newline at end of file
+.venv
```
Comment (Ivanmaxx) [inline .gitignore:7]: why'd you delete and add it back tho

File: backend/shared_library/db.py [hunk 2]
```diff
@@ -16,26 +16,33 @@
 from shared_library import utils
 import typing
 
-class DB_connection(utils.singleton):
+class DB_connection(utils.Singleton):
```
Comment (Ivanmaxx) [inline backend/shared_library/db.py:19]: stop doing that lol

File: backend/shared_library/utils/singleton.py [hunk 3]
```diff
@@ -19,7 +19,8 @@ class Singleton:
 
     _instance = None
 
-    def __new__(class_, *args, **kwargs):
+    # TODO: remove arg1. Currently I'm not sure how to handle this
```
Comment (Ivanmaxx) [inline backend/shared_library/utils/singleton.py:22]: Why is a todo there? shouldn't the code be done lol

**Review by Ivanmaxx — APPROVED** [review]
Good i guess, you defo coded better than i did

File: backend/users/models.py
Comment (M-Stepan-373) [inline backend/users/models.py:1]: Many metods havent docs
fix it

File: .gitignore (hunk 1 above)
Comment (Minuta18) [reply .gitignore:7]: Is it same files?

File: backend/shared_library/db.py (hunk 2 above)
Comment (Minuta18) [reply backend/shared_library/db.py:19]: I changed `singleton` to `Singleton`

File: backend/shared_library/utils/singleton.py (hunk 3 above)
Comment (Minuta18) [reply backend/shared_library/utils/singleton.py:22]: `*args` and `**kwargs` could be removed in future, but I dont know how

**Comment by M-Stepan-373** [comment]
Duck you
qa3wse4dr5tfgyhujimkl

**Comment by M-Stepan-373** [comment]
Muck you

**Comment by M-Stepan-373** [comment]
Cluck you

**Review by M-Stepan-373 — APPROVED** [review]

**Pull request merged**